contrib/pkg/clusterpool/clusterclaim.go

```diff
@@ -0,0 +1,106 @@
+package clusterpool
+
+import (
+	"fmt"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	"github.com/openshift/hive/contrib/pkg/utils"
+	"github.com/openshift/hive/pkg/apis"
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+)
+
+type ClusterClaimOptions struct {
+	Namespace       string
+	Lifetime        time.Duration
+	ClusterPoolName string
+
+	homeDir string
+}
+
+func NewClaimClusterPoolCommand() *cobra.Command {
+	opt := &ClusterClaimOptions{}
+
+	opt.homeDir = "."
+
+	defaultDuration, _ := time.ParseDuration("8h")
+
+	cmd := &cobra.Command{
+		Use:   "claim CLUSTER_POOL_NAME",
+		Short: "claim a cluster from a clusterpool",
+		Long:  "claims a cluster from the clusterpool in the given namespace",
+		Args:  cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			log.SetLevel(log.InfoLevel)
+			opt.ClusterPoolName = args[0]
+			err := opt.run()
+			if err != nil {
+				log.WithError(err).Fatal("Error")
+			}
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVarP(&opt.Namespace, "namespace", "n", "",
+		"Namespace to create cluster claim in. Has to be the namespace in which the cluster pool is deployed")
+	flags.DurationVar(&opt.Lifetime, "lifetime", defaultDuration, "Lifetime of the cluster claim. "+
```

Comment:
I would have expected that the default would be to not set a lifetime.

Reply:
Agreed.